include/nccl_ofi.h

```diff
@@ -357,66 +357,105 @@ struct nccl_net_ofi_device {
  * generally it is expected that calls into resources that share the
  * same domain will share the same lock.
  */
-struct nccl_net_ofi_domain {
-	/* Backpointer to the device associated with this domain. */
-	nccl_net_ofi_device_t *device;
+class nccl_net_ofi_domain_t {
+public:
+	/**
+	 * @brief	Default constructor.
+	 * 
+	 * Initialize resources associated with the domain base class.
+	 * Expectation is that this will be called by a transport's domain
+	 * constructor 
+	 */	
+	nccl_net_ofi_domain_t(nccl_net_ofi_device_t *device_arg, long creating_thread_id);
+	
+	/**
+	 * Retrieve an fid_domain object associated with this domain. There may
+	 * be more than one fid_domain per domain, depending on the transport;
+	 * in that case, this will be the domain object associated with the
+	 * "leader NIC"
+	 */
+	virtual struct fid_domain *get_ofi_domain() = 0;
+
+	/**
+	 * Retrieve an fid_cq object associated with this domain. There may be
+	 * more than one cq per domain, depending on the transport; in that
+	 * case, this will be the info object associated with the "leader NIC"
+	 */
+	virtual struct fid_cq *get_ofi_cq() = 0;
+
+	/* Create a new endpoint
+	 *
+	 * Pure virtual function to allocate a new endpoint structure
+	 */
+	virtual nccl_net_ofi_ep_t *create_endpoint() = 0;
+
+	/**
+	 * @brief	Returns the base domain's device back-pointer.
+	 */
+	inline nccl_net_ofi_device_t *get_base_device()
```

Comment:
should just be `get_device()`, no?

Reply:
Fixed in latest force push to rename to `get_device`. I was originally thinking to differentiate between a base domain's `get_base_device` function and the SENDRECV/RDMA transport domain equivalents for getting transport-specific device types (`sendrecv_domain_get_device` and `rdma_domain_get_device`), but from the base domain's perspective it should just be `get_device`.